aea/configurations/base.py

```diff
@@ -197,6 +198,7 @@ def __init__(self,
         self.authors = authors
         self.version = version
         self.license = license
+        self.uuid = generate_uuid(author=self.authors, package_name=self.name, version=self.version)
```

Comment:
@marcofavorito and @Totoual
- are we happy calling it uuid? It makes sense, but maybe not quite in line with crypto lingo.
- we should allow for a fourth, optional field in the `generate_uuid` function called `nonce`. Then we can by default set the nonce to None. If the user wants to use two packages with different configs then they can achieve that by feeding a `nonce` to the `generate_uuid` function. Note, we do not allow for the submission of the same package with two different nonces to the package registry, hence we do not need to keep track of the nonce in the config (it is only for local usage).
- I think we need to add a `user_name` field and replace the authors with the `user_name`
- All packages should have `user_name = fetchai`
- All packages should have their uuid in the config (only do this once we have finished everything else int his PR)
- Once this PR is finished, we need another PR which applies the usage of uuid to how packages are identified within the AEA framework, but we cannot do this for 2.0, hence please open an issue.